src/libs/auth.ts

```diff
@@ -55,8 +68,12 @@ export const getOidcConfig = () => {
   };
 };
 
-const getAuthInstance = () => {
-  return authStore.get().authContext;
+const getAuthInstance = (): AuthContextProps => {
+  const authContext: AuthContextProps | undefined = authStore.get().authContext;
+  if (authContext === undefined) {
+    throw new Error('Auth instance is trying to be accessed before it is initialized');
```

Comment:
I think this error could benefit from a little word smithing to help guide a developer who encounters this towards resolving it.

This is what I came up with:
```suggestion
    throw new Error('getAuthInstance must not be called before authContext is initialized.');
```

Reply:
My understanding was that if we throw an Error that it will propagate up to the user, so this was written to be seen by users. Is that correct? In that case, I could add the more detailed message as a console.error message.

Reply:
In this case, I think we expect this error to only ever be encountered by developers. If it's thrown, it's revealing a bug and means that this function was used in a place where it shouldn't be.

Its main purpose is to satisfy TypeScript that `authStore.get().authContext` is initialized when this function is called. That it is currently used in production without anything checking if it returns undefined strongly suggests that that assumption is a valid.

Reply:
I agree that this should only be encountered by developers, but just to err on the side of caution I made the Error thrown be in more plain language and added a console.error with a more descriptive message.

Reply:
Sounds good.